projects/addon-mobile/components/mobile-calendar/mobile-calendar.component.ts

```diff
@@ -139,17 +136,34 @@ export class TuiMobileCalendarComponent {
     }
 
     ngAfterViewInit() {
-        // Virtual scroll has not yet rendered items even in ngAfterViewInit
         this.monthsScrollRef?.scrolledIndexChange
             .pipe(take(1), takeUntil(this.destroy$))
             .subscribe(() => {
-                this.scrollToActiveYear();
-                this.scrollToActiveMonth();
                 this.initYearScroll();
                 this.initMonthScroll();
+
+                if (this.getYearsViewportSize() > 0) {
+                    this.scrollToActiveYearAndMonth();
+                } else {
+                    this.idleCheckViewportSize();
+                }
             });
     }
 
+    getYearsViewportSize() {
+        return this.yearsScrollRef?.getViewportSize() || 0;
+    }
+
+    checkViewportSize() {
+        this.yearsScrollRef?.checkViewportSize();
```

Comment:
What do these methods do and why we call them before scrolling?

Reply:
We call them so that the virtual scroll recalculates the dimensions relative to its parent element

Reply:
fixed